### ISSUE TITLE
panic when accessing index concurrently with go 1.11

### ISSUE THREAD

**Issue by Funzinator** [blevesearch/bleve#993]
Hi,

When using go 1.11, I get `panic`s and the application crashes when doing concurrent access to the Bleve index. I tried to work around that by using a mutex in front of my operations, but that didn't help.

The same code is working fine with go 1.10.4. I am not sure if I should open this issue here or at the golang repo.

What I am doing:
- one goroutine walks through all the documents in the database and calls `index.Index(id, doc)` (not in batches).
- another goroutine searches at the same time for particular documents.

This is the panic output I get. Maybe you've seen something like this already and can point me in the right direction.

Thanks,
Funzi

```runtime: bad pointer in frame funzi.org/index/indexer.(*BleveIndex).SearchActor at 0xc000e35550: 0x2
fatal error: invalid pointer found on stack

runtime stack:
runtime.throw(0xa698a4, 0x1e)
	/usr/local/go/go-1.11/src/runtime/panic.go:608 +0x72 fp=0x7f4dd3de1750 sp=0x7f4dd3de1720 pc=0x42c812
runtime.adjustpointers(0xc000e35348, 0x7f4dd3de1808, 0x7f4dd3de1bc8, 0xeb0998, 0xecfa60)
	/usr/local/go/go-1.11/src/runtime/stack.go:591 +0x221 fp=0x7f4dd3de17b0 sp=0x7f4dd3de1750 pc=0x4436c1
runtime.adjustframe(0x7f4dd3de1ad8, 0x7f4dd3de1bc8, 0xecfa60)
	/usr/local/go/go-1.11/src/runtime/stack.go:633 +0x188 fp=0x7f4dd3de1838 sp=0x7f4dd3de17b0 pc=0x443858
runtime.gentraceback(0xffffffffffffffff, 0xffffffffffffffff, 0x0, 0xc000000300, 0x0, 0x0, 0x7fffffff, 0xa83ab8, 0x7f4dd3de1bc8, 0x0, ...)
	/usr/local/go/go-1.11/src/runtime/traceback.go:325 +0x1326 fp=0x7f4dd3de1b40 sp=0x7f4dd3de1838 pc=0x44daf6
runtime.copystack(0xc000000300, 0x4000, 0x20300000000001)
	/usr/local/go/go-1.11/src/runtime/stack.go:845 +0x26e fp=0x7f4dd3de1cf8 sp=0x7f4dd3de1b40 pc=0x443f0e
runtime.newstack()
	/usr/local/go/go-1.11/src/runtime/stack.go:1017 +0x312 fp=0x7f4dd3de1e90 sp=0x7f4dd3de1cf8 pc=0x444322
runtime.morestack()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:429 +0x8f fp=0x7f4dd3de1e98 sp=0x7f4dd3de1e90 pc=0x456acf

goroutine 1 [copystack]:
regexp/syntax.(*compiler).compile(0xc000e34a50, 0xc00015c1c0, 0x7ffffff)
	/usr/local/go/go-1.11/src/regexp/syntax/compile.go:98 +0x1832 fp=0xc000e34528 sp=0xc000e34520 pc=0x6b49d2
regexp/syntax.(*compiler).compile(0xc000e34a50, 0xc00015c2a0, 0xc000e9a840)
	/usr/local/go/go-1.11/src/regexp/syntax/compile.go:142 +0xe14 fp=0xc000e34790 sp=0xc000e34528 pc=0x6b3fb4
regexp/syntax.(*compiler).compile(0xc000e34a50, 0xc00015c620, 0x0)
	/usr/local/go/go-1.11/src/regexp/syntax/compile.go:154 +0xadf fp=0xc000e349f8 sp=0xc000e34790 pc=0x6b3c7f
regexp/syntax.Compile(0xc00015c620, 0xc00015c620, 0x1, 0x1)
	/usr/local/go/go-1.11/src/regexp/syntax/compile.go:83 +0x105 fp=0xc000e34ac0 sp=0xc000e349f8 pc=0x6b2f85
regexp.compile(0xc000684620, 0x9, 0xc0000000d4, 0x7, 0xc000684620, 0x9)
	/usr/local/go/go-1.11/src/regexp/regexp.go:178 +0xc7 fp=0xc000e34b48 sp=0xc000e34ac0 pc=0x6cc2c7
regexp.Compile(0xc000684620, 0x9, 0x7, 0xc000684620, 0x9)
	/usr/local/go/go-1.11/src/regexp/regexp.go:133 +0x41 fp=0xc000e34b88 sp=0xc000e34b48 pc=0x6cc1c1
funzi.org/vendor/github.com/blevesearch/bleve/search/query.(*WildcardQuery).convertToRegexp(0xc000c56030, 0xc000237d50, 0x4, 0xc00063e1c0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/search/query/wildcard.go:105 +0x5b fp=0xc000e34bc0 sp=0xc000e34b88 pc=0x8f63db
funzi.org/vendor/github.com/blevesearch/bleve/search/query.(*WildcardQuery).Searcher(0xc000c56030, 0xb04c00, 0xc00063e1c0, 0xb02d20, 0xc00014ef80, 0x0, 0xb03980, 0xc0012fc700, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/search/query/wildcard.go:88 +0x100 fp=0xc000e34c38 sp=0xc000e34bc0 pc=0x8f6250
funzi.org/vendor/github.com/blevesearch/bleve/search/query.(*ConjunctionQuery).Searcher(0xc000c56060, 0xb04c00, 0xc00063e1c0, 0xb02d20, 0xc00014ef80, 0x7f0000, 0xc00014ef58, 0xc00063e1c0, 0xc00014ef58, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/search/query/conjunction.go:58 +0x122 fp=0xc000e34cf0 sp=0xc000e34c38 pc=0x8e63b2
funzi.org/vendor/github.com/blevesearch/bleve/search/query.(*BooleanQuery).Searcher(0xc000e50000, 0xb04c00, 0xc00063e1c0, 0xb02d20, 0xc00014ef80, 0xc000f50000, 0x0, 0x0, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/search/query/boolean.go:132 +0x410 fp=0xc000e34d90 sp=0xc000e34cf0 pc=0x8e58d0
funzi.org/vendor/github.com/blevesearch/bleve.(*indexImpl).SearchInContext(0xc000242770, 0xb00b00, 0xc000024128, 0xc00015c070, 0x0, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index_impl.go:391 +0x2a9 fp=0xc000e35268 sp=0xc000e34d90 pc=0x8fdf09
funzi.org/vendor/github.com/blevesearch/bleve.(*indexImpl).Search(0xc000242770, 0xc00015c070, 0xf3b440, 0x8, 0xc000110180)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index_impl.go:363 +0x4d fp=0xc000e352b0 sp=0xc000e35268 pc=0x8fdc2d
funzi.org/index/indexer.(*BleveIndex).SearchActor(0xc000275100, 0xa58697, 0x5, 0xc00090c000, 0x268b, 0x3000, 0xafe8e0, 0xc000275100)
	/home/funzi/go/src/funzi.org/index/indexer/bleve.go:130 +0x9ca fp=0xc000e355b0 sp=0xc000e352b0 pc=0x914b2a
main.main()
	/home/funzi/go/src/funzi.org/index/main.go:196 +0x144a fp=0xc000e35f98 sp=0xc000e355b0 pc=0x91797a
runtime.main()
	/usr/local/go/go-1.11/src/runtime/proc.go:201 +0x207 fp=0xc000e35fe0 sp=0xc000e35f98 pc=0x42e187
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000e35fe8 sp=0xc000e35fe0 pc=0x458a61

goroutine 2 [force gc (idle)]:
runtime.gopark(0xa83cf8, 0xf3ab70, 0x1410, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000048f80 sp=0xc000048f60 pc=0x42e57b
runtime.goparkunlock(0xf3ab70, 0x1410, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:308 +0x53 fp=0xc000048fb0 sp=0xc000048f80 pc=0x42e623
runtime.forcegchelper()
	/usr/local/go/go-1.11/src/runtime/proc.go:251 +0xb3 fp=0xc000048fe0 sp=0xc000048fb0 pc=0x42e3f3
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000048fe8 sp=0xc000048fe0 pc=0x458a61
created by runtime.init.4
	/usr/local/go/go-1.11/src/runtime/proc.go:240 +0x35

goroutine 3 [GC sweep wait]:
runtime.gopark(0xa83cf8, 0xf3aea0, 0x45140c, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000049780 sp=0xc000049760 pc=0x42e57b
runtime.goparkunlock(0xf3aea0, 0xaf140c, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:308 +0x53 fp=0xc0000497b0 sp=0xc000049780 pc=0x42e623
runtime.bgsweep(0xc000072000)
	/usr/local/go/go-1.11/src/runtime/mgcsweep.go:71 +0x102 fp=0xc0000497d8 sp=0xc0000497b0 pc=0x421412
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0000497e0 sp=0xc0000497d8 pc=0x458a61
created by runtime.gcenable
	/usr/local/go/go-1.11/src/runtime/mgc.go:216 +0x58

goroutine 4 [finalizer wait]:
runtime.gopark(0xa83cf8, 0xf590e8, 0x140f, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000049f28 sp=0xc000049f08 pc=0x42e57b
runtime.goparkunlock(0xf590e8, 0xa8140f, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:308 +0x53 fp=0xc000049f58 sp=0xc000049f28 pc=0x42e623
runtime.runfinq()
	/usr/local/go/go-1.11/src/runtime/mfinal.go:175 +0x99 fp=0xc000049fe0 sp=0xc000049f58 pc=0x418d79
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000049fe8 sp=0xc000049fe0 pc=0x458a61
created by runtime.createfing
	/usr/local/go/go-1.11/src/runtime/mfinal.go:156 +0x61

goroutine 5 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000161dc8 sp=0xc000161da8 pc=0x42e57b
runtime.selectgo(0xc000161f70, 0xc000161f50, 0x2, 0x1, 0xc000048701)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000161f28 sp=0xc000161dc8 pc=0x43d7dd
funzi.org/vendor/github.com/blevesearch/bleve/index.AnalysisWorker(0xc0000342a0, 0xc000034300)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:83 +0x102 fp=0xc000161fd0 sp=0xc000161f28 pc=0x78cc62
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000161fd8 sp=0xc000161fd0 pc=0x458a61
created by funzi.org/vendor/github.com/blevesearch/bleve/index.NewAnalysisQueue
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:75 +0xc8

goroutine 6 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000e9ddc8 sp=0xc000e9dda8 pc=0x42e57b
runtime.selectgo(0xc000e9df70, 0xc000e9df50, 0x2, 0x1, 0x1)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000e9df28 sp=0xc000e9ddc8 pc=0x43d7dd
funzi.org/vendor/github.com/blevesearch/bleve/index.AnalysisWorker(0xc0000342a0, 0xc000034300)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:83 +0x102 fp=0xc000e9dfd0 sp=0xc000e9df28 pc=0x78cc62
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000e9dfd8 sp=0xc000e9dfd0 pc=0x458a61
created by funzi.org/vendor/github.com/blevesearch/bleve/index.NewAnalysisQueue
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:75 +0xc8

goroutine 7 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000e99dc8 sp=0xc000e99da8 pc=0x42e57b
runtime.selectgo(0xc000e99f70, 0xc000e99f50, 0x2, 0x1, 0x1)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000e99f28 sp=0xc000e99dc8 pc=0x43d7dd
funzi.org/vendor/github.com/blevesearch/bleve/index.AnalysisWorker(0xc0000342a0, 0xc000034300)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:83 +0x102 fp=0xc000e99fd0 sp=0xc000e99f28 pc=0x78cc62
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000e99fd8 sp=0xc000e99fd0 pc=0x458a61
created by funzi.org/vendor/github.com/blevesearch/bleve/index.NewAnalysisQueue
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:75 +0xc8

goroutine 8 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000167dc8 sp=0xc000167da8 pc=0x42e57b
runtime.selectgo(0xc000167f70, 0xc000167f50, 0x2, 0x1, 0x1)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000167f28 sp=0xc000167dc8 pc=0x43d7dd
funzi.org/vendor/github.com/blevesearch/bleve/index.AnalysisWorker(0xc0000342a0, 0xc000034300)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:83 +0x102 fp=0xc000167fd0 sp=0xc000167f28 pc=0x78cc62
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000167fd8 sp=0xc000167fd0 pc=0x458a61
created by funzi.org/vendor/github.com/blevesearch/bleve/index.NewAnalysisQueue
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/analysis.go:75 +0xc8

goroutine 9 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc00004bdd8 sp=0xc00004bdb8 pc=0x42e57b
runtime.selectgo(0xc00004bf68, 0xc00004bf60, 0x2, 0x0, 0x0)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc00004bf38 sp=0xc00004bdd8 pc=0x43d7dd
database/sql.(*DB).connectionOpener(0xc000110480, 0xb00ac0, 0xc000070dc0)
	/usr/local/go/go-1.11/src/database/sql/sql.go:1001 +0xe8 fp=0xc00004bfc8 sp=0xc00004bf38 pc=0x4ddb08
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00004bfd0 sp=0xc00004bfc8 pc=0x458a61
created by database/sql.OpenDB
	/usr/local/go/go-1.11/src/database/sql/sql.go:671 +0x15d

goroutine 10 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc0000445c0 sp=0xc0000445a0 pc=0x42e57b
runtime.selectgo(0xc000044768, 0xc000044748, 0x2, 0x1, 0x1)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000044720 sp=0xc0000445c0 pc=0x43d7dd
database/sql.(*DB).connectionResetter(0xc000110480, 0xb00ac0, 0xc000070dc0)
	/usr/local/go/go-1.11/src/database/sql/sql.go:1014 +0xfb fp=0xc0000447c8 sp=0xc000044720 pc=0x4ddc3b
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0000447d0 sp=0xc0000447c8 pc=0x458a61
created by database/sql.OpenDB
	/usr/local/go/go-1.11/src/database/sql/sql.go:672 +0x193

goroutine 13 [select]:
runtime.gopark(0xa83d38, 0x0, 0x1809, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000045530 sp=0xc000045510 pc=0x42e57b
runtime.selectgo(0xc0000456f0, 0xc0000456c4, 0x2, 0x0, 0x0)
	/usr/local/go/go-1.11/src/runtime/select.go:313 +0xcbd fp=0xc000045690 sp=0xc000045530 pc=0x43d7dd
funzi.org/vendor/github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher.func1(0xc0000750e0, 0xc000110540, 0xc000034660)
	/home/funzi/go/src/funzi.org/vendor/github.com/go-sql-driver/mysql/connection_go18.go:179 +0xbf fp=0xc0000457c8 sp=0xc000045690 pc=0x7139bf
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0000457d0 sp=0xc0000457c8 pc=0x458a61
created by funzi.org/vendor/github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher
	/home/funzi/go/src/funzi.org/vendor/github.com/go-sql-driver/mysql/connection_go18.go:176 +0xbe

goroutine 14 [GC worker (idle)]:
runtime.gopark(0xa83ba8, 0xc0004511d0, 0x1417, 0x0)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000044f60 sp=0xc000044f40 pc=0x42e57b
runtime.gcBgMarkWorker(0xc000036000)
	/usr/local/go/go-1.11/src/runtime/mgc.go:1772 +0xfc fp=0xc000044fd8 sp=0xc000044f60 pc=0x41c64c
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000044fe0 sp=0xc000044fd8 pc=0x458a61
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/go-1.11/src/runtime/mgc.go:1720 +0x77

goroutine 15 [GC worker (idle)]:
runtime.gopark(0xa83ba8, 0xc000468000, 0x1417, 0x0)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc000045f60 sp=0xc000045f40 pc=0x42e57b
runtime.gcBgMarkWorker(0xc000038500)
	/usr/local/go/go-1.11/src/runtime/mgc.go:1772 +0xfc fp=0xc000045fd8 sp=0xc000045f60 pc=0x41c64c
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000045fe0 sp=0xc000045fd8 pc=0x458a61
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/go-1.11/src/runtime/mgc.go:1720 +0x77

goroutine 18 [GC worker (idle)]:
runtime.gopark(0xa83ba8, 0xc000472000, 0x1417, 0x0)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc00046e760 sp=0xc00046e740 pc=0x42e57b
runtime.gcBgMarkWorker(0xc00003aa00)
	/usr/local/go/go-1.11/src/runtime/mgc.go:1772 +0xfc fp=0xc00046e7d8 sp=0xc00046e760 pc=0x41c64c
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00046e7e0 sp=0xc00046e7d8 pc=0x458a61
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/go-1.11/src/runtime/mgc.go:1720 +0x77

goroutine 34 [GC worker (idle)]:
runtime.gopark(0xa83ba8, 0xc000472010, 0x1417, 0x0)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc00046a760 sp=0xc00046a740 pc=0x42e57b
runtime.gcBgMarkWorker(0xc00003cf00)
	/usr/local/go/go-1.11/src/runtime/mgc.go:1772 +0xfc fp=0xc00046a7d8 sp=0xc00046a760 pc=0x41c64c
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00046a7e0 sp=0xc00046a7d8 pc=0x458a61
created by runtime.gcBgMarkStartWorkers
	/usr/local/go/go-1.11/src/runtime/mgc.go:1720 +0x77

goroutine 16 [runnable]:
sync.(*Pool).Put(0xc00018c330, 0x971c00, 0xc0006478e0)
	/usr/local/go/go-1.11/src/sync/pool.go:88 +0x19f fp=0xc000e393c8 sp=0xc000e393c0 pc=0x46f59f
funzi.org/vendor/github.com/boltdb/bolt.(*Tx).write(0xc000226e00, 0x4be7c478a, 0xf3b440)
	/home/funzi/go/src/funzi.org/vendor/github.com/boltdb/bolt/tx.go:538 +0x45d fp=0xc000e39510 sp=0xc000e393c8 pc=0x8965bd
funzi.org/vendor/github.com/boltdb/bolt.(*Tx).Commit(0xc000226e00, 0xc000f57900, 0xc000e39678)
	/home/funzi/go/src/funzi.org/vendor/github.com/boltdb/bolt/tx.go:198 +0x310 fp=0xc000e39650 sp=0xc000e39510 pc=0x894470
funzi.org/vendor/github.com/blevesearch/bleve/index/store/boltdb.(*Writer).ExecuteBatch.func1(0xc000e397f0, 0xc000226e00)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/store/boltdb/writer.go:52 +0x4e fp=0xc000e39678 sp=0xc000e39650 pc=0x8b165e
funzi.org/vendor/github.com/blevesearch/bleve/index/store/boltdb.(*Writer).ExecuteBatch(0xc00000f0c0, 0xb01600, 0xc00096ff00, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/store/boltdb/writer.go:90 +0x503 fp=0xc000e397d8 sp=0xc000e39678 pc=0x8b11b3
funzi.org/vendor/github.com/blevesearch/bleve/index/upsidedown.(*UpsideDownCouch).batchRows(0xc00014ef00, 0xb00f40, 0xc00000f0c0, 0x0, 0x0, 0x0, 0xc00096fe80, 0x1, 0x1, 0x0, ...)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/upsidedown/upsidedown.go:292 +0x78e fp=0xc000e39c98 sp=0xc000e397d8 pc=0x7f075e
funzi.org/vendor/github.com/blevesearch/bleve/index/upsidedown.(*UpsideDownCouch).Update(0xc00014ef00, 0xc0013d9600, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index/upsidedown/upsidedown.go:484 +0x511 fp=0xc000e39e28 sp=0xc000e39c98 pc=0x7f2351
funzi.org/vendor/github.com/blevesearch/bleve.(*indexImpl).Index(0xc000242770, 0xc00075d908, 0x5, 0x9b5fe0, 0xc000226b60, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/vendor/github.com/blevesearch/bleve/index_impl.go:252 +0x21e fp=0xc000e39ea8 sp=0xc000e39e28 pc=0x8fd34e
funzi.org/index/indexer.(*BleveIndex).IndexActor(0xc000275100, 0xc00000ef30, 0x1, 0x1, 0x0, 0x0)
	/home/funzi/go/src/funzi.org/index/indexer/bleve.go:158 +0xa5 fp=0xc000e39ef8 sp=0xc000e39ea8 pc=0x9150d5
main.main.func1(0xc000237d70, 0xc00090c000, 0x268b, 0x3000, 0xafe8e0, 0xc000275100, 0xc000240e60)
	/home/funzi/go/src/funzi.org/index/main.go:177 +0x17a fp=0xc000e39fa8 sp=0xc000e39ef8 pc=0x91845a
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000e39fb0 sp=0xc000e39fa8 pc=0x458a61
created by main.main
	/home/funzi/go/src/funzi.org/index/main.go:170 +0x140e

goroutine 50 [timer goroutine (idle)]:
runtime.gopark(0xa83cf8, 0xf3f900, 0x1414, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:302 +0xeb fp=0xc00046cf28 sp=0xc00046cf08 pc=0x42e57b
runtime.goparkunlock(0xf3f900, 0xc000001414, 0x1)
	/usr/local/go/go-1.11/src/runtime/proc.go:308 +0x53 fp=0xc00046cf58 sp=0xc00046cf28 pc=0x42e623
runtime.timerproc(0xf3f900)
	/usr/local/go/go-1.11/src/runtime/time.go:280 +0x288 fp=0xc00046cfd8 sp=0xc00046cf58 pc=0x44a1d8
runtime.goexit()
	/usr/local/go/go-1.11/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00046cfe0 sp=0xc00046cfd8 pc=0x458a61
created by runtime.(*timersBucket).addtimerLocked
	/usr/local/go/go-1.11/src/runtime/time.go:170 +0x114
```

**Comment by steveyen** [comment]
Thx for the report.  I, for one, have never seen an issue like this before.

> The same code is working fine with go 1.10.4. I am not sure if I should open this issue here or at the golang repo.

Indeed - I'd recommend getting a report for the golang team.  The runtime.morestack() thing looks at first glance like some deep golang internals thing.

**Comment by Funzinator** [comment]
It was indeed a bug in the Go compiler that will be fixed in Go 1.11.1 and Go 1.12 onwards.
Closing it here. Thanks!

**Comment by steveyen** [comment]
> It was indeed a bug in the Go compiler that will be fixed in Go 1.11.1 and Go 1.12 onwards.

Thanks for following up on that and the update.  Link to the go 1.11 ticket - https://github.com/golang/go/issues/27278
